apps/web-giddh/src/app/reports/components/payment-report/payment-report.component.ts

```diff
@@ -0,0 +1,762 @@
+        );
+    }
+
+    /**
+     * This will hide the datepicker
+     *
+     * @memberof AuditLogsFormComponent
+     */
+    public hideGiddhDatepicker(): void {
+        this.modalRef.hide();
+    }
+
+    /**
+     * Call back function for date/range selection in datepicker
+     *
+     * @param {*} value
+     * @memberof AuditLogsFormComponent
+     */
+    public dateSelectedCallback(value?: any): void {
+        if (value && value.event === "cancel") {
+            this.hideGiddhDatepicker();
+            return;
+        }
+        this.selectedRangeLabel = "";
+
+        if (value && value.name) {
+            this.selectedRangeLabel = value.name;
+        }
+        this.hideGiddhDatepicker();
+        if (value && value.startDate && value.endDate) {
+            this.selectedDateRange = { startDate: moment(value.startDate), endDate: moment(value.endDate) };
+            this.selectedDateRangeUi = moment(value.startDate).format(GIDDH_NEW_DATE_FORMAT_UI) + " - " + moment(value.endDate).format(GIDDH_NEW_DATE_FORMAT_UI);
+            this.fromDate = moment(value.startDate).format(GIDDH_DATE_FORMAT);
+            this.toDate = moment(value.endDate).format(GIDDH_DATE_FORMAT);
+            this.showClearFilter = true;
+            this.fetchPaymentsData();
+        }
+    }
+
+    /**
+     * Callback for translation response complete
+     *
+     * @param {boolean} event
+     * @memberof PaymentReportComponent
+     */
+    public translationComplete(event: boolean): void {
+        if (event) {
+            this.paymentAdvanceSearchAmountFilters = [
+                { label: this.commonLocaleData?.app_comparision_filters.greater_than, value: 'GREATER_THAN' },
+                { label: this.commonLocaleData?.app_comparision_filters.greater_than_equals, value: 'GREATER_THAN_OR_EQUALS' },
+                { label: this.commonLocaleData?.app_comparision_filters.less_than_equals, value: 'LESS_THAN_OR_EQUALS' },
+                { label: this.commonLocaleData?.app_comparision_filters.equals, value: 'EQUALS' },
+                { label: this.commonLocaleData?.app_comparision_filters.not_equals, value: 'NOT_EQUALS' }
+            ];
+            this.advanceSearchModel.totalAmountFilter.filterValues = this.paymentAdvanceSearchAmountFilters;
+            this.advanceSearchModel.unusedAmountFilter.filterValues = this.paymentAdvanceSearchAmountFilters;
+        }
+    }
+
+    /**
+     * This will update the search query param
+     *
+     * @param {*} value
+     * @memberof PaymentReportComponent
+     */
+    public updateSearchQuery(value): void {
+        this.searchQueryParams.q = value;
+    }
+
+    /**
+     * This will preview voucher
+     *
+     * @param {*} payment
+     * @memberof PaymentReportComponent
+     */
+    public previewVoucher(payment: any): void {
+        // if (this.voucherApiVersion === 2) {
```

Comment:
⚠️ Empty method definition

Reply:
yes we are not allowing preview/edit currently.